assets/js/components/utils/index.js

```diff
@@ -107,3 +107,24 @@ export const getCategory = ( categoryId ) => {
 		path: `${ ENDPOINTS.categories }/${ categoryId }`,
 	} );
 };
+
+/**
+ * Get a promise that resolves to an array of category objects from the API.
+ */
+export const getCategories = () => {
+	return apiFetch( {
+		path: addQueryArgs( `${ ENDPOINTS.products }/categories`, { per_page: -1 } ),
+	} );
+};
```

Comment:
Not needed in this pull, but with the number of blocks calling product categories, this seems like a good thing to move to a wp.data store so that blocks using categories can share what is retrieved.  This will help with limiting the number of queries to the server (same for product attributes and product attribute terms).  That's something I can do in a pull?

Reply:
Sure, sounds like a good idea! :slightly_smiling_face: